imqsauth/http.go

```diff
@@ -84,9 +88,19 @@ type ImqsCentral struct {
 	Config    *Config
 	Central   *authaus.Central
 	Yellowfin *Yellowfin
+
+	// ChangeSubscribers maintains a list of web socket connections for
+	// pushing role changes of connected users to their browsers.
+	// Currently no communication is expected to be received FROM the browser.
+	// On error or EOF the connection is closed and removed from the map.
+	RoleChangeSubscribers map[uint64]*websocket.Conn
+
+	subscriberId uint64
+	sync.RWMutex
```

Comment:
It's a bad idea to make the entire ImqsCentral inherit from a lock (RWMutex in this case).
Why?
Because it makes it very unclear what the lock is guarding. The number one rule of locks is that it is crystal clear what the lock guards. By making the entire ImqsCentral a locking object, one is implicitly saying that any state within ImqsCentral needs to be read/written under that lock. This is definitely not true, and if it was true, then the entire auth system would effectively turn into a single threaded beast. A reader of the code most definitely will not know under which conditions he needs to take this mysterious lock.

So.. the lock needs to be a regular member variable, with a clear name and comment, such as `subscriberLock sync.RWMutex`. And the comment should include in detail, everything that this lock guards, such as "Guards access to RoleChangeSubscribers and subscriberId". I would place the lock above the members that it guards.

`RoleChangeSubscribers` cannot be allowed to be public. Firstly, by making a member public, the user won't know that this object is guarded by a lock. And secondly, the lock is private, so even if he wanted to do the right thing, he wouldn't be able to!

In my (little) world, it's customary to name something like `subscriberId` `nextSubscriberId`. The way it is used right now, it is `lastSubscriberId`. I guess either way is fine, so long as the name reflects it's usage.